test/regression/expected/query_filter.out

```diff
@@ -28,3 +28,51 @@ SELECT COUNT(*) FROM query_filter_float WHERE a < 1.1;
 (1 row)
 
 DROP TABLE query_filter_float;
+CREATE TABLE query_filter_varchar(a VARCHAR);
+INSERT INTO query_filter_varchar VALUES ('t1'), ('t2'), ('t1');
+SELECT COUNT(*)FROM query_filter_varchar WHERE a = 't1';
+ count 
+-------
+     2
+(1 row)
+
+SELECT COUNT(a) FROM query_filter_varchar WHERE a = 't1';
+ count 
+-------
+     2
+(1 row)
+
+SELECT a, COUNT(*) FROM query_filter_varchar WHERE a = 't1' GROUP BY a;
+ a  | count 
+----+-------
+ t1 |     2
+(1 row)
+
+DROP TABLE query_filter_varchar;
+CREATE TABLE query_filter_output_column(a INT, b VARCHAR, c FLOAT8);
+INSERT INTO query_filter_output_column VALUES (1, 't1', 1.0), (2, 't1', 2.0), (2, 't2', 1.0);
+-- Projection ids list will be used (column `a`is not needed after scan)
+SELECT b FROM query_filter_output_column WHERE a = 2;
+ b  
+----
+ t1
+ t2
+(2 rows)
+
+-- Column ids list used because both of fetched column are used after scan
+SELECT a, b FROM query_filter_output_column WHERE b = 't1';
```

Comment:
From your comments and my closer reading of the relevant duckdb code, we are expected to change the handle the column order changes. So let's include that in one of these tests
```suggestion
-- Column ids list used because both of fetched column are used after scan, we also should swap column order correctly
SELECT b, a FROM query_filter_output_column WHERE b = 't1';
```

Reply:
Let's also add a test where we only re-order columns, but still output all of them.